index.d.ts

```diff
@@ -626,3 +626,49 @@ interface KVNamespace {
     cursor: string;
   }>;
 }
+
+interface DurableObjectEntries<T> {
+  [key: string]: T;
+}
+
+interface DurableObjectListOptions {
+  start: string;
+  end: string;
+  reverse: boolean;
+  limit: number;
+}
+
+interface DurableObjectStorage {
+  get<ExpectedValue = unknown>(key: string): Promise<ExpectedValue>;
+  get<ExpectedValue = unknown>(keys: Array<string>): Promise<Map<string, ExpectedValue>>;
+  put<GivenValue = unknown>(key: string, value: GivenValue): Promise<void>;
+  put<GivenValue = unknown>(entries: DurableObjectEntries<GivenValue>): Promise<void>;
+  delete(key: string): Promise<boolean>;
+  delete(keys: Array<string>): Promise<boolean>;
+  list<ExpectedValue = unknown>(
+    options?: DurableObjectListOptions
+  ): Promise<Map<string, ExpectedValue>>;
+  transaction<ExpectedValue = unknown>(
+    closure: (txn: DurableObjectStorage) => Promise<void>
+  ): Promise<Map<string, ExpectedValue>>;
+}
+
+interface DurableObjectState {
```

Comment:
Might be more pain than it's worth, but it's a little odd to have an interface for one of the values passed to the constructor for a durable object, and not an interface for the constructor itself.

Reply:
I would add the constructor but I don't think you can include a constructor in a typescript interface. I think the DurableObjectState type will still be useful for people implementing the class

Reply:
I'm definitely not a typescript guru, but couldn't that be an abstract class?  Might still not be worth it since the second parameter to the constructor could be essentially anything depending on environment bindings.

Agree that the DurableObjectState type is useful either way

Reply:
On further thought, I'd leave this as is for now.  Abstract class will work, but I think in the near future we may want some actual behavior people can extend, e.g. to take care of initialization from durable storage.  Can wait til then.

Reply:
The abstract class would be extended, so it wouldn't actually enforce the constructor signature. It does require a `super` call to the abstract class but I'm not sure that has value.